src/OrbitBase/NotFoundOrTest.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest-death-test.h>
+#include <gtest/gtest.h>
+
+#include "OrbitBase/NotFoundOr.h"
+
+namespace orbit_base {
+
+TEST(NotFoundOr, IsNotFound) {
```

Comment:
In addition to `int` and `void` can you also add tests for some move-only type? Maybe `std::unique_ptr<int>`?

Reply:
done